nautobot/extras/models/models.py

```diff
@@ -196,7 +197,24 @@ def get_config_context(self):
             config_context_data = ConfigContext.objects.get_for_object(self).values_list("data", flat=True)
         else:
             config_context_data = self.config_context_data or []
+            # Device and VirtualMachine's Location has its own ConfigContext and its parent Locations' ConfigContext, if any, should
+            # also be applied. However, since moving from mptt to django-tree-queries https://github.com/nautobot/nautobot/issues/510,
+            # we lost the ability to query the ancestors for a particular tree node for subquery https://github.com/matthiask/django-tree-queries/issues/54.
+            # So instead of constructing the location related query in ConfigContextModelQueryset._get_config_context_filters(), which is complicated across databases
+            # We append the missing parent location query here as a patch.
```

Comment:
Lets make we document this in `ConfigContextModelQueryset._get_config_context_filters()`

Reply:
Did that in this commit https://github.com/nautobot/nautobot/pull/4417/commits/24f41c70e1ccfc7934c384f89c1d63078179b18d